src/app/components/hris/employees/employee-profile/accordions/accordion-career-summary/accordion-career-summary-qualifications/accordion-career-summary-qualifications.component.ts

```diff
@@ -0,0 +1,241 @@
+        this.sharedAccordionFunctionality.employeeQualificationDto.school = data.school;
+        this.sharedAccordionFunctionality.employeeQualificationDto.fieldOfStudy = data.fieldOfStudy;
+        this.sharedAccordionFunctionality.employeeQualificationDto.proofOfQualification = data.proofOfQualification;
+        this.sharedAccordionFunctionality.employeeQualificationDto.documentName = data.documentName;
+        this.proofOfQualificationFinal = data.proofOfQualification
+        console.log(this.sharedAccordionFunctionality.employeeQualificationDto);
+        if (data.year && data.year.endsWith("-01-01")) {
+          this.sharedAccordionFunctionality.employeeQualificationDto.year = data.year.substring(0, 4);
+        } else {
+          this.sharedAccordionFunctionality.employeeQualificationDto.year = data.year;
+        }
+        this.sharedAccordionFunctionality.employeeQualificationDto.nqfLevel = data.nqfLevel;
+        this.initializeForm();
+      },
+      error: (error) => {
+        this.snackBarService.showSnackbar(error.error, "snack-error");
+      }
+    })
+  }
+
+  onFileChange(event: any): void {
+    if (event.target.files && event.target.files.length) {
+      this.fileUploaded = true;
+      const file = event.target.files[0];
+      this.fileName = file.name;
+      this.sharedAccordionFunctionality.employeeQualificationDto.documentName = file.name;
+      if (this.validateFile(file)) {
+        this.fileConverter(file, 'proofOfQualification');
+      }
+    }
+  }
+
+  validateFile(file: File): boolean {
+    const allowedTypes = ['application/pdf'];
+    if (!allowedTypes.includes(file.type)) {
+      this.isValidFile = false;
+      return false;
+    }
+    if (file.size > 10 * 1024 * 1024) {
+      this.isValidFileSize = false;
+      return false;
+    }
+    this.isValidFileSize = true;
+    return true;
+  }
+
+  fileConverter(file: File, controlName: string) {
+    const reader = new FileReader();
+    reader.addEventListener('loadend', () => {
+      const base64Data = reader.result as string;
+      this.sharedAccordionFunctionality.employeeQualificationForm.patchValue({ [controlName]: base64Data });
+      this.proofOfQualificationFinal = base64Data;
+    });
+    reader.readAsDataURL(file);
+  }
+
+  downloadFile(base64String: string, fileName: string) {
+    const commaIndex = base64String.indexOf(',');
+    if (commaIndex !== -1) {
+      base64String = base64String.slice(commaIndex + 1);
+    }
+    const byteString = atob(base64String);
+    const arrayBuffer = new ArrayBuffer(byteString.length);
+    const intArray = new Uint8Array(arrayBuffer);
+
+    for (let i = 0; i < byteString.length; i++) {
+      intArray[i] = byteString.charCodeAt(i);
+    }
+
+    const blob = new Blob([arrayBuffer], { type: 'application/pdf' });
+    const link = document.createElement('a');
+    link.href = window.URL.createObjectURL(blob);
+    link.download = fileName;
```

Comment:
This would not be necessary as the fileName parameter is already specified as a string type. Thus using the template literal creates no real benefit. I personally don't see it as a cleaner syntax approach unless used to build a string.